pyclean/modern.py

```diff
@@ -88,9 +88,25 @@ def descend_and_clean_bytecode(directory):
             if child.suffix in ['.pyc', '.pyo']:
                 Runner.unlink(child)
         elif child.is_dir():
-            descend_and_clean_bytecode(child)
+            if child.name == '.pytest_cache' and clean_test_cache:
+                descend_and_remove_directory(child)
+            else:
+                descend_and_clean_bytecode(child, clean_test_cache)
 
             if child.name == '__pycache__':
                 Runner.rmdir(child)
         else:
             log.debug("Ignoring %s", child)
+
+
+def descend_and_remove_directory(directory):
+    """
+    Delete all files in a directory, then delete the directory.
+    """
+    for child in directory.iterdir():
+        if child.is_file():
+            Runner.unlink(child)
+        elif child.is_dir():
+            descend_and_remove_directory(child)
+
+    Runner.rmdir(directory)
```

Comment:
Isn't that the same as [shutil.rmtree](https://docs.python.org/3/library/shutil.html#shutil.rmtree)?
```python
shutil.rmtree(directory)
```

Reply:
`rmdir` will throw an error if there are any files remaining in the directory. This way, the file/directory count displayed at the end are correct, including test cache files.